src/shrpx.cc

```diff
@@ -1088,6 +1111,13 @@ int call_daemon() {
 #ifdef __sgi
   return _daemonize(0, 0, 0, 0);
 #else  // !__sgi
+#ifdef HAVE_LIBSYSTEMD
+  if (sd_booted()) {
+    LOG(NOTICE) << "Daemonising disabled under systemd";
+    chdir("/");
+    return 0;
+  }
```

Comment:
If not started by systemd, $NOTIFY_SOCKET is not set, so sd_notifyf() returns without doing anything.
If not booted with systemd, function sd_booted() in line 1115 returns false and daemon() is called.